### ISSUE TITLE
[Chapter 2] Exercise 4 prepare and predict pipeline

### ISSUE THREAD

**Issue by cooke99** [ageron/handson-ml2#168]
Not sure if this is the correct place for this question, but is there any point in practice to a pipeline that both prepares the data and then makes the prediction, as in Exercise 4? 

From what I understand in the book, a pipeline causes fit_transform to be called for each transformer on the data sequentially - does this not mean that if you were to use this pipeline for the test set, it would fit the transformers and model to the test set, which is not what we want?

**Comment by Rosseel** [comment]
You are correct in your assumption but you glanced over a piece of text on page 79 -2nd edition on 
'Evaluate your system on the Test Set'
"run your full_pipeline to transform the data (call transform() , not fit_transform() - you do not want to fit the test set!)"
